src/main/webapp/search_area.js

```diff
@@ -135,13 +147,69 @@ class SearchArea {
     this.organizationObjectsList = await response.json();
   }
 
-  async setUrlParamValue(urlParamKey, urlParamValue) {
-    /* if the param is a zipcode, replace any existing one. Otherwise, add it to existing params */    
+  async setUrlParamValue(urlParamKey, urlParamValue) {   
+    /* New query value is not added if it is a duplicate or empty/null */
+    if (this.filterParams.getAll("filterParam").includes(urlParamValue) ||
+        this.filterParams.getAll("zipcode").includes(urlParamValue) ||
+        (urlParamValue === null) || (urlParamValue.trim() === "")) {
+      return;
+    }
+    
+    /* if the param is a zipcode, remove tag of any existing one & set new one*/ 
     if (urlParamKey === "zipcode") {
```

Comment:
Being nitpicky for future you:

This looks good in general, but consider if you want a new param that you want to have this single-value-only type semantics.

What if you had a Set that described which parameters you wanted as singleton

Set<String> singletonParameters = ....  right now just "zipcode"

Then a lot of these functions will read:

if (singletonParameters.contains(urlParamKey)) {  ... }

Then you can add new singleton parameters easily.